action/find_save_tools.py

```diff
@@ -1,52 +1,46 @@
-""" this contains useful functions for loading and saving data tables"""
+import itertools
 import os
-from pathlib import PurePosixPath
+import pathlib
+from typing import Dict, Iterable, Optional, Sequence, Union
 
 import pandas as pd
 
 from .errors import ImportActionError
 
+TableConfig = Dict[str, Union[str, Sequence[str]]]
 
-def import_data(variable_json, data="output/input.csv"):
-    """
-    Imports data and checks that the variables are present
 
-    Will accept input file as csv or dta file (stata).
-    """
-    # load the data into a pandas DataFrame depending on ext
-    ext = PurePosixPath(data).suffix
-    if ext == ".csv":
-        df = pd.read_csv(data)
-    elif ext == ".gz":
-        df = pd.read_csv(data, compression="gzip")
-    elif ext == ".dta":
-        df = pd.read_stata(data)
-    elif ext == ".feather":
-        df = pd.read_feather(data)
+def import_data(file_path: pathlib.Path, table_configs: Dict[str, TableConfig]):
+    """Imports data, checking that the required variables are present."""
+    if file_path.suffix == ".csv":
+        table = pd.read_csv(file_path)
+    elif file_path.suffix == ".gz":
+        table = pd.read_csv(file_path, compression="gzip")
+    elif file_path.suffix == ".dta":
+        table = pd.read_stata(file_path)
+    elif file_path.suffix == ".feather":
+        table = pd.read_feather(file_path)
     else:
-        raise ImportActionError("Unsupported filetype attempted to be imported")
+        raise ImportActionError(f"'{file_path.suffix}' is not a supported file-type")
 
-    # checks that the variables defined in the json are column names in the
-    # csv and raises an error if note
-    for table_names, instructions in variable_json.items():
-        if not set(instructions["variables"]).issubset(df.columns.values):
-            raise ImportActionError
+    variables = set(itertools.chain(*[x["variables"] for x in table_configs.values()]))
```

Comment:
`itertools` is powerful and terse: hopefully you understand what's going on here. Working from the inside out:

* Iterate over each item in `table_configs.values()`, which is a list (more-or-less 🙂), returning the contents of the `variables` property, which is a list. This gives us a list of lists.
* Expand (`*`) the list of lists. [`itertools.chain`](https://docs.python.org/3.8/library/itertools.html#itertools.chain) receives multiple lists as multiple positional arguments. It iterates over elements from the first list until this list is exhausted; then the second list; then the third list; etc.
* Reduce the list of variables to a set of variables (i.e. no duplicates)